tutorials/02_flask.md

```diff
@@ -0,0 +1,352 @@
+Soubor `requirements.txt` se závislostmi:
+
+```
+Flask
+requests
+click
+```
+
+A soubor `wsgi.py`, který slouží jako vstupní soubor pro OpenShift.
+V něm je třeba importovat naší aplikaci jako `application`.
+Zde předpokládáme, že soubor s aplikací se jmenuje `hello.py`.
+
+```python
+from hello import app as application
+```
+
+Nové commity pushněte na OpenShift:
+
+    git push openshift master
+
+Pokud jste při vytváření aplikace nepřidali váš repozitář,
+možná budete muset použít sílu:
+
+    git push --force openshift master
+
+To je vše, aplikace by měla běžet na
+[flask-hroncok.rhcloud.com](https://flask-hroncok.rhcloud.com/) (či podobně).
+Nezapomeňte nové commity pushovat do hlavního repozitáře s kódem
+(na Githubu nebo GiLabu) i na OpenShift:
+
+    git push  # pushuje do původního repozitáře
+    git push openshift master
+
+[OpenShift]: https://www.openshift.com/
+[OpenShift Web Console]: https://openshift.redhat.com/app/console/applications
+
+### API klíče na OpenShiftu
+
+Protože vaše tajné klíče nejsou v repozitáři, nabízí se otázka, jak je předat
+OpenShiftu, aby o nich věděl. Kromě přístupu přes git můžete k aplikaci
+přistupovat i přes ssh. Konfigurační soubor s API klíči tak můžete nakopírovat
+do adresáře k tomu určenému (ten je soukromý, vidíte ho jen vy):
+
+    scp auth.cfg 123456789123456789123456@flask-hroncok.rhcloud.com:app-root/data/
+
+Z aplikace k němu přistoupíte třeba takto:
+
+```python
+import os
+
+authfile = 'auth.cfg'
+
+if 'OPENSHIFT_DATA_DIR' in os.environ:
+    authfile = os.path.join(os.environ['OPENSHIFT_DATA_DIR'], authfile)
+```
+
+Stejnou složku použijte, bude-li vaše aplikace potřebovat nějaké trvalé
+úložiště pro zapisování dat (databázím se zde věnovat nebudeme).
+
+**Poznámka:** Doporučujeme pro tyto potřeby stejně raději nepoužívat API klíče
+k vlastním účtům, raději si vyrobte nějaké účty pouze pro tento účel.
+
+Úkol
+----
+
+Vaším úkolem za 5 bodů je vytvořit rozšířit command line aplikaci z minulého
+cvičení o webové rozhraní. Stávající funkcionalita ale musí být zachována,
+k tomu můžete použít například podpříkazy pro click:
+
+```python
+@click.group()
+def cli():
+    pass
+
+@cli.command()
+def wev():
+    """Run the web app"""
+    click.echo('Running the web app')
+
+@cli.command()
+def console():
+    """Run the console app"""
+    click.echo('Running the console app')
+```
+
+Výslednou aplikaci nasaďte na OpenShift.
+Odkaz na běžící aplikaci a repozitář nám pošlete e-mailem.
+V repozitáři prosím nastavte tag `v0.2`.
+Termín odevzdání je začátek příštího cvičení (dřívější paralelky).
+
+### Twitter Wall
+
+Konzole není pro Twitter Wall dostatečně vhodné médium,
+doplňte do aplikace webový frontend, který bude zobrazovat avatary, obrázky,
+odkazy... Hledaný pojem by měl jít zadat pomocí URL.
+
+Pokud to neumíte nebo nechce, netrapte se s JavaScriptem, ale po načtení
+stránky prostě zobrazte několik posledních tweetů.
```

Comment:
Jasný. Základy HTML ale předpokládám na magisterském umí.

Co takhle:

Konzole není pro Twitter Wall dostatečně vhodné médium,
doplňte do aplikace webový frontend, který bude zobrazovat
výsledky hledání. Hledaný pojem by měl jít zadat pomocí URL.

Pro plný počet bodů  musí rozhraní zobrazovat avatary uživatelů a zpracovávat [entity] jako obrázky, odkazy a hash tagy.

[entity]: https://dev.twitter.com/overview/api/entities-in-twitter-objects